pkg/results/error.go

```diff
@@ -33,23 +33,30 @@ func (e *Error) Is(target error) bool {
 	return is
 }
 
-// FullReason provides the chain of error reasons, divided by colons
-func (e *Error) FullReason() string {
-	reasonedError := &Error{}
-	if !errors.As(e.wrapped, &reasonedError) {
-		return string(e.reason)
+// Reasons provides the chains of error reasons.
+// Each item in the return value is a single chain divided by colons.  Aggregate
+// errors — those whose type provides an `Errors` method returning a list of
+// errors — are recursively expanded, generating a separate chain for each
+// child.
+func Reasons(errs ...error) (ret []string) {
```

Comment:
Did we lose the unknown return>

Reply:
My initial implementation kept it, but in the end I moved it [here](https://github.com/openshift/ci-tools/pull/1989/files#diff-d89ae970e1b23b3fe7065c1c228c08354612fe0ed0cc69db9dd5565dd16987b5R111) because the API made more sense to me and it simplified the recursive implementation of this function.